src/data/polynomial/degree/definitions.lean

```diff
@@ -222,6 +222,13 @@ begin
   { rw [← C_mul_X_pow_eq_monomial, nat_degree_C_mul_X_pow i r hr] }
 end
 
+lemma nat_degree_monomial_eq (i : ℕ) {r : R} (r0 : r ≠ 0) :
```

Comment:
Perhaps you should add `[decidable_eq R]` to match `nat_degree_monomial`?

Reply:
I thought that if you can get away with using classical in the proof, you should not assume decidable stuff.  Maybe I'm wrong, though: I'm not so much into decidability.

Reply:
You could technically get away with using `classical` anywhere. It's just that some people have decided that some things are worth making constructive. I'm no expert on this either, but I presume that consistency would be the best option here.

Reply:
I agree: classical everywhere would be my preferred solution!

Reply:
There are some rare cases where computability can be convenient for non-computable mathematicians. It's really nice when you're able to prove an equality by literally just letting the compiler compute the values for you.

Reply:
The problem with classical everywhere is that it doesn't mean "don't care about classicality", it means "use the decidability instance given by the axiom of choice". In particular if sometime we use a more concrete decidability instance (for example we have it for `ℕ`, and this is very good, so we can prove things like `1 ∈ {1, 2}` using `rfl`) this doesn't match with the one given by choice, and we are in trouble.